nucleus/payara-modules/hazelcast-bootstrap/src/main/java/fish/payara/nucleus/hazelcast/HazelcastCore.java

```diff
@@ -555,6 +561,26 @@ public int getPort() {
 
     @Override
     public UnprocessedChangeEvents changed(PropertyChangeEvent[] pces) {
-        return null;
+        List<UnprocessedChangeEvent> unprocessedChanges = new ArrayList<>();
+        for (PropertyChangeEvent pce : pces) {
+            if (pce.getPropertyName().equalsIgnoreCase("datagrid-encryption-enabled")) {
+                unprocessedChanges.add(new UnprocessedChangeEvent(pce, "Hazelcast encryption settings changed"));
+            }
+        }
+
+        if (unprocessedChanges.isEmpty()) {
+            return null;
+        }
+        return new UnprocessedChangeEvents(unprocessedChanges);
+    }
+
+    public boolean isDatagridEncryptionEnabled() {
+        // If the datagrid encryption settings have changed, we want to return the opposite of the current setting
+        // to prevent the server changing encryption behaviour without a restart
+        boolean datagridEncryptionEnabled = configuration.getDatagridEncryptionEnabled().equalsIgnoreCase("true");
+        if (oldDatagridEncryptionValue != datagridEncryptionEnabled) {
+            return oldDatagridEncryptionValue;
+        }
+        return datagridEncryptionEnabled;
```

Comment:
If `oldDatagridEncryptionValue` is returned in case it is different to `datagridEncryptionEnabled` this method can be simplified to just `return oldDatagridEncryptionValue`.

Reply:
🤦‍♂️